WordPress/src/test/java/org/wordpress/android/ui/activitylog/BackupDownloadStatusServiceTest.kt

```diff
@@ -0,0 +1,301 @@
+            backupDownloadStatusFetchError = it }
+        whenever(activityLogStore.getBackupDownloadStatusForSite(site)).thenReturn(null)
+        whenever(activityLogStore.fetchBackupDownloadState(any())).thenReturn(
+                OnBackupDownloadStatusFetched(FETCH_BACKUP_DOWNLOAD_STATE)
+        )
+        whenever(activityLogStore.backupDownload(any())).thenReturn(
+                OnBackupDownload(
+                        rewindId = rewindId,
+                        causeOfChange = BACKUP_DOWNLOAD
+                )
+        )
+        whenever(activityLogStore.getActivityLogItemByRewindId(rewindId)).thenReturn(
+                activityLogModel
+        )
+        whenever(backupDownloadProgressChecker.startNow(any(), any())).thenReturn(null)
+    }
+
+    @After
+    fun tearDown() {
+        backupDownloadStatusService.stop()
+    }
+
+    @Test
+    fun `emits available BackupDownloadStatus on start when download not in progress`() {
+        whenever(activityLogStore.getBackupDownloadStatusForSite(site)).thenReturn(
+                inProgressBackupDownloadStatusModel.copy(progress = null)
+        )
+
+        backupDownloadStatusService.start(site)
+
+        assertEquals(backupDownloadAvailable, true)
+    }
+
+    @Test
+    fun `emits unavailable BackupDownloadStatus on start when download is in progress`() = runBlocking {
+        val inactiveBackupDownloadStatusModel = inProgressBackupDownloadStatusModel
+        whenever(activityLogStore.getBackupDownloadStatusForSite(site)).thenReturn(
+                inactiveBackupDownloadStatusModel,
+                null
+        )
+
+        backupDownloadStatusService.start(site)
+
+        assertEquals(backupDownloadAvailable, false)
+    }
+
+    @Test
+    fun `triggers fetch when BackupDownloadStatus not available`() = runBlocking {
+        backupDownloadStatusService.start(site)
+
+        assertFetchBackupDownloadStatusAction()
+    }
+
+    @Test
+    fun `updates BackupDownloadStatus and restarts checker when BackupDownload not already running`() =
+            runBlocking<Unit> {
+                backupDownloadStatusService.start(site)
+                whenever(activityLogStore.getBackupDownloadStatusForSite(site)).thenReturn(
+                        inProgressBackupDownloadStatusModel
+                )
+                whenever(activityLogStore.fetchBackupDownloadState(any())).thenReturn(
+                        OnBackupDownloadStatusFetched(FETCH_BACKUP_DOWNLOAD_STATE)
+                )
+                reset(backupDownloadProgressChecker)
+
+                backupDownloadStatusService.requestStatusUpdate()
+
+                verify(backupDownloadProgressChecker).startNow(
+                        site,
+                        inProgressBackupDownloadStatusModel.downloadId
+                )
+            }
+
+    @Test
+    fun `triggers BackupDownload And Makes Action Unavailable`() = runBlocking {
```

Comment:
Typo - capital letters: **A**nd **M**akes **A**ction **U**navailable

Reply:
Thanks - updated in d7d59e8